### PR TITLE
Indices::push(u32)

### PR DESCRIPTION
**Pull request by stepancheg**
# Objective

Making work with `Indices` struct easier. Currently when building indices in some quick-and-dirty code we need to do matches and handle enum variants.

## Solution

`Indices::push` utility which works transparently with `U16` and `U32` variants.

## Testing

Unit test added.

### CODE REVIEW

**Comment by stepancheg** [comment]
Also, there's silent integer overflow in this code:

https://github.com/bevyengine/bevy/blob/74dedb284143c2af8b8aeacf4a3c653aa74b7e74/crates/bevy_mesh/src/mesh.rs#L851-L853

This code could benefit from changes like this (but not this directly).

File: crates/bevy_mesh/src/index.rs [hunk 1]
```diff
@@ -135,3 +155,28 @@ impl From<&Indices> for IndexFormat {
...
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use crate::Indices;
+    use wgpu::IndexFormat;
+
+    #[test]
+    fn test_indices_push() {
```
Comment (alice-i-cecile) [inline crates/bevy_mesh/src/index.rs:165]: Can you add a test case or test for the u16 -> u32 behavior, since that's particularly important to get right?

Comment (stepancheg) [reply crates/bevy_mesh/src/index.rs:165]: The test is there. Code below:

```
        indices.push(0x10000);
        assert_eq!(IndexFormat::Uint32, IndexFormat::from(&indices));
        assert_eq!(vec![10, 0x10000], indices.iter().collect::<Vec<_>>());
```

converts `Indices` from `U16` to `U32`. I'll add a comment to make it more clear.

**Review by alice-i-cecile — APPROVED** [review]

**Review by IceSentry — APPROVED** [review]
I'm not sure if I like the auto conversion, but it's not a blocker for me.

**Pull request merged**